src/services/aws/aws.ts

```diff
@@ -139,30 +126,40 @@ export const {
   },
 } = aws;
 
-type EndowCardFields = keyof (Omit<EndowmentCard, "hq" | "categories"> &
-  /** replace with cloudsearch specific field format */
-  Pick<EndowmentProfileUpdate, "hq_country" | "categories_sdgs">);
-
-//object format first to avoid duplicates
-const endowCardObj: {
-  [key in EndowCardFields]: any; //we care only for keys
-} = {
-  hq_country: "",
-  active_in_countries: "",
-  categories_sdgs: "",
-  id: "",
-  image: "",
-  kyc_donors_only: "",
-  name: "",
-  tagline: "",
-  endow_type: "",
+export const useEndowmentsQuery = <T extends Partial<Endowment>>(
+  qParams: EndowmentsQueryParams<T>
+) => {
+  const result = aws.endpoints.endowments.useQuery(qParams);
+  return {
+    ...result,
+    data: result.data as PaginatedAWSQueryRes<T[]>,
+    originalArgs: result.originalArgs as EndowmentsQueryParams<T>,
+  };
 };
-const endowCardFields = Object.keys(endowCardObj).join(",");
 
-const ENDOW_ID_NAME_OBJ: {
-  [key in Extract<EndowCardFields, "id" | "name">]: any;
-} = {
-  id: "",
-  name: "",
+export const useLazyEndowmentsQuery = () => {
+  const [fetch, ...rest] = aws.endpoints.endowments.useLazyQuery();
+  const func = <T extends Partial<Endowment>>(
+    params: EndowmentsQueryParams<T>
+  ) =>
+    fetch(params).then((res) => ({
+      ...res,
+      data: res.data as PaginatedAWSQueryRes<T[]>,
+    }));
+  return [func, ...rest] as [typeof func, ...typeof rest];
 };
-const ENDOW_ID_NAME_FIELDS = Object.keys(ENDOW_ID_NAME_OBJ).join(",");
+
+function createReturnParam(templateResult: Partial<Endowment>): string {
+  const { categories, ...okFields } = templateResult;
+
+  const cloudSearchParams: EndowmentCloudSearchFields = {
+    ...okFields,
```

Comment:
> this is just adding additional layers of code
> instead of using the library directly

This PR introduces a new level of abstraction to allow reusing the same endpoint while maintaining type-safety, maintaining performance and maintaining the amount of data transferred over the network. Having an extra 50 lines of code to remove `useEndowmentCardsQuery` (doesn't return "endowment cards" but returns just "endowments") and `useEndowmentIdNameQuery` (ridiculous ad-hoc name), both of which are placed in the root `aws` service, but are used in only one specific place.

@SovereignAndrey @ap-justin

Reply:
> `Endowment` and `EndowmentCloudSearchFields` are two separate types - we can't just merge one to the other just because it happened to be Partial

They are defined as completely separate types https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/generic-endow-query-v2/src/types/aws/ap/index.ts. Are you having a problem with the spread operator?

@ap-justin

Reply:
> the gains aren't justified

What do you mean "aren't justified"? What have we been discussing then this whole time if not **justifications** "for" and "against"?

> it would have been easier if this future usecase is something we see

Sometimes it's ok to change existing things, so that no suboptimal precedents are set.

> it's still easier to just append another builder.query

Easier than using an existing one?

> and simply transfers Cards and Names projection to the component

Yes, it "simply" does that. It transfers the responsibility for stating which fields are required on the UI to the component that actually has to display that UI. **<-- This is the whole point of the PR**

> Code is just shuffled

Not seeing the point of this, that's what refactoring usually entails.